### ISSUE TITLE
Improve/fix indent size detection for multiline doc-comments

### ISSUE THREAD

**Issue by beeb** [helix-editor/helix#9822]
### Summary

Solidity (and likely other languages) allows for two types of documentation comment styles:

```solidity
/// @notice This is the default one-line comment style

/**
 * @notice And this is the multi-line doc-comment style
 */
```

While the first style gives the expected indent guides and doesn't affect indent-size detection, the second is somewhat problematic for some files.

### Reproduction Steps

Consider the following file:

```solidity
// SPDX-License-Identifier: UNLICENSED
pragma solidity >=0.8.0;

/**
 * @title MyInterface
 * @notice Defines some events
 */
interface MyInterface {
    /**
     * @notice My comment
     */
    event SomeEvent();
}
```

When opened in helix, it interprets the indent size as being 1 space.

![image](https://github.com/helix-editor/helix/assets/703631/478f6c75-e99a-4d7d-9309-c47aebda6bba)

### Helix log

N/A

### Platform

Linux (Ubuntu)

### Terminal Emulator

wezterm 20240203-110809-5046fc22

### Installation Method

nixpkgs

### Helix Version

helix 23.10 (f6021dd0)

**Comment by pascalkuthe** [comment]
Thus would probably be solved by #9556 for a small example like this. Large files basically are never made of of more comments than actual source lines.

There is no way to detect comments in the indent heuristic so I don't think we can do more than that closing as duplicate of #9556

**Comment by beeb** [comment]
Hopefully this will fix it, but note that I'm seeing the same problem on very large files, so long as they start like the example above.
